### PR TITLE
Build on Mac

### PR DESCRIPTION
**Pull request by robinbb**

### CODE REVIEW

**Comment by robinbb** [comment]
@facundominguez I addressed the unusual Git history by redo-ing this issue. It now builds on Mac.

**Comment by robinbb** [comment]
@facundominguez Ping.

**Comment by robinbb** [comment]
If anyone has time to launch a build, please verify this issue. I'll add others to the assignees list.

File: apps/argv/argv.cabal [hunk 1]
```diff
@@ -13,4 +13,5 @@ executable sparkle-example-argv
   build-depends: base >=4.7 && <5, distributed-closure, sparkle, text, vector
   default-language: Haskell2010
   ghc-options: -dynamic -lpthread -threaded
-  ld-options: -pie -Wl,-z,origin -Wl,-rpath,$ORIGIN
+  if !os(darwin)
+    ld-options: -pie -Wl,-z,origin -Wl,-rpath,$ORIGIN
```
Comment (facundominguez) [inline apps/argv/argv.cabal]: When user docker `!os(darwin)` is always true. Are you trying to build the examples outside docker?

**Comment by facundominguez** [comment]
`stack --nix install` works for me.

**Pull request closed without merging**